src/sns-handler.js

```
@@ -0,0 +1,36 @@
+import LambdaHandler from './lambda-handler';
+
+export default class SnsHandler extends LambdaHandler {
+  static SnsError(error) {
+    if (!(error instanceof Error)) {
+      error = new Error(error);
+    }
+
+    return error;
+  }
+
+  _processEvent(event) {
+    let initialRecordSns;
+
+    try {
+      initialRecordSns = event.Records[0].Sns;
+    } catch (err) {
+      throw new SnsHandler.SnsError('Sns handler attempted to process an event with incorrect format');
+    }
+
+    return {
+      subject: initialRecordSns.Subject,
+      message: initialRecordSns.Message,
+      timeStamp: initialRecordSns.Timestamp
+    };
+  }
+
+  _processResponse() {
+    return 'Success';
+  }
+
+  _processError() {
+    return 'Failure';
```

This is not a change request, but I just want to document a hunch, unless it's already been squashed, that this should actually throw an error, which would then be passed as the first argument to the lambda callback. We chose not to do this with HTTP because it provided no feedback to the client making the request, but in this case, since the caller is SNS, it may be valuable to actually send an error.
```js
_processError(err) {
  throw err;
}
```